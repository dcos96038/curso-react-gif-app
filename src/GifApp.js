import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifApp = ({ defaultCategories = [] }) => {
  // const [categories, setCategories] = useState(["Messi"]);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1>Gif APP</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};

export default GifApp;
