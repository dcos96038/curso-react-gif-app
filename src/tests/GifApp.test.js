import { shallow } from "enzyme";
import { GifApp } from "../GifApp";

describe("Pruebas en el componente GifApp", () => {
  test("El componente debe mostrarse correctamente", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("El componente GifGrid debe aparecer en una cantidad igual a la longitud de la lista de categorias", () => {
    const categories = ["messi", "ronaldo"];
    const wrapper = shallow(<GifApp defaultCategories={categories} />);

    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
