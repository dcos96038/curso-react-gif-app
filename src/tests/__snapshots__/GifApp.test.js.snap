// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifApp El componente debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Gif APP
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
